### PR TITLE
revamp docs and create clear install and first use section

### PR DESCRIPTION
**Pull request by nslater**

### CODE REVIEW

File: docs/compatibility.txt [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+.. rubric:: Table of Contents
+
+.. contents::
+   :local:
+
+JDBC
+====
+
+This JDBC driver follows the JDBC 4.1 standard as it uses PGSQL driver as the
```
Comment (andreidan) [inline docs/compatibility.txt]: This is a bit off as we are not using the pgsql driver but  we speak the postgres wire protocol. Other than that it's our own implementation

File: docs/compatibility.txt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+compatible with CrateDB.
+
+**The following features are not supported:**
+
+* ``java.sql.ParameterMetaData`` as returned by e.g. ``java.sql.PreparedStatement``
+* ``DataSource``
+* ``CallableStatement`` as CrateDB does not support stored procedures
+
+For further details about compatibility with all possible JDBC features,
+see the ``ResultSet``, ``ResultSetMetaData`` and ``DatabaseMetaData``
```
Comment (andreidan) [inline docs/compatibility.txt]: links to the implementations

File: docs/compatibility.txt [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+CrateDB
+=======
+
+All versions of the JDBC driver >= 2.0.x require CrateDB 0.56.0 or newer.
+Previous versions of the JDBC driver (1.x) are only compatible with older
+versions of CrateDB.
+
+The support for versions of the JDBC driver >= 2.0.x with CrateDB version
+0.56.x is **experimental**. The binding between versions of the JDBC
+driver >= 2.0.x and CrateDB will be stable when 0.57.x of CrateDB is stable.
```
Comment (andreidan) [inline docs/compatibility.txt]: 0.57.x is probably stable for a long time (and if it's not, we don't care about it now)

File: docs/compatibility.txt (hunk 1 above)
Comment (nslater) [reply docs/compatibility.txt]: fixed

File: docs/compatibility.txt (hunk 2 above)
Comment (nslater) [reply docs/compatibility.txt]: can you provide the links? wasn't able to find what I am supposed to link to

File: docs/compatibility.txt (hunk 3 above)
Comment (nslater) [reply docs/compatibility.txt]: fixed?

**Comment by nslater** [comment]
@andreidan can you recheck please?

File: docs/compatibility.txt (hunk 2 above)
Comment (matriv) [reply docs/compatibility.txt]: hm I don't know what andrei means here, let's wait for him...

Comment (matriv) [reply docs/compatibility.txt]: Probably: https://github.com/crate/pgjdbc/blob/77a2d396dfe8ec89081894f8d0150a85129c6bd0/pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java
https://github.com/crate/pgjdbc/blob/77a2d396dfe8ec89081894f8d0150a85129c6bd0/pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSetMetaData.java and https://github.com/crate/pgjdbc/blob/77a2d396dfe8ec89081894f8d0150a85129c6bd0/pgjdbc/src/main/java/org/postgresql/jdbc/PgDatabaseMetaData.java

Comment (andreidan) [reply docs/compatibility.txt]: I was thinking of links to these interfaces we mention: 
- https://docs.oracle.com/javase/7/docs/api/java/sql/ResultSet.html
- https://docs.oracle.com/javase/7/docs/api/java/sql/ResultSetMetaData.html
- https://docs.oracle.com/javase/7/docs/api/java/sql/DatabaseMetaData.html

**Comment by nslater** [comment]
@andreidan can you recheck please?

**Pull request merged**